client/api/go-client/client.go

```diff
@@ -355,3 +360,30 @@ func (c *ClientOptions) retryDelay(r *http.Response) time.Duration {
 	s := rand.Intn(max-min) + min
 	return time.Second * time.Duration(s)
 }
+
+// CheckRedirectFunc is an alias for the somewhat complex function signature
+// of the CheckRedirect function of the http.Client.
+type CheckRedirectFunc func(*http.Request, []*http.Request) error
+
+// ClientFunc is an alias for the function signature needed to create custom
+// http clients.
+type ClientFunc func(*tls.Config, CheckRedirectFunc) HttpPerformer
+
+// HttpPerformer is an interface that the heketi api client needs from the http
+// client.
+type HttpPerformer interface {
```

Comment:
This is an especially useful aspect to this PR, because it assures callers that only `.Do()` will be used from the http.Client (or whatever) that is passed in.